ocs_ci/helpers/disruption_helpers.py

```diff
@@ -23,6 +24,7 @@ class Disruptions:
     selector = None
     daemon_pid = None
 
+    @run_on_cluster_type(cluster_type="provider")
```

Comment:
What if `Disruptions` constructor method will set the cluster's details, to run the commands on, based on whether or not the relevant resources exist in it? 
For instance, it will check on which cluster the ODF pods exist and once it finds the cluster, which should be the provider, it will use its details within the `Disruptions` class's methods

Reply:
This possibility was tested when trying out different ways of implementation. Switching to provider context will make other simultaneous commands (run on consumer) to fail. Passing arguments to the called functions are also tested.
eg:
```
self.resource_obj[resource_id].delete(force=True, cluster=self.cluster_config)
```
after making the delete command to accept additional argument (testes key word args also). Doing this across ocs-ci tests will need code changes in many places(within test cases, helpers, utils etc). All intermediate functions will need to accept the cluster argument an pass it until reaching the function which runs the command.
I have also tried setting cluster argument after creating the OCP object of kind pod.
```
eg:  
setattr(self.resource_obj, cluster, <cluster conf or kubeconfig>)
```
This function `run_on_cluster_type` is not just created for the mentioned tier4 tests. This can be used in other functions which need provider cluster.

Reply:
This could be solved by creating locks on oc command calls and making sure that oc command calls in other threads wait until their turn and are not executed at the same time when other call is active. I think that this can be accomplished by using Lock objects (https://docs.python.org/3/library/threading.html#lock-objects) in functions that handle command execution (maybe exec_cmd).

Reply:
@fbalak The requirement is to run two or more threads together when needed. So one thread waiting for another won't help in such cases.